src/deploy/lifecycleHooks.ts

```diff
@@ -68,7 +67,12 @@ function getChildEnvironment(target, overallOptions, config) {
   });
 }
 
-function runTargetCommands(target, hook, overallOptions, config) {
+function runTargetCommands(
```

Comment:
Optional, but tons of the code in this file could be rewritten to use async/await instead of manually manipulating promise

Reply:
yup, but not the goal of this PR. :)